sml_small/editing/thousand_pounds/thousand_pounds.py

```diff
@@ -76,24 +119,50 @@ def run(
     ) as error:  # Catch any underlying errors and return a coherent output dataset
         # Ensure we populate the output target variables with the same output values as originally given
         target_variables_final = []
-        for question in target_variables:
-            target_variables_final.append(
-                Target_variable(
-                    identifier=question.identifier,
-                    original_value=question.original_value,
-                    final_value=question.original_value,
-                )
-            )
-
-        return Thousands_output(
```

Comment:
Was this the equivalent of returning a STOP marker?  We may need to talk through if this should have been removed.

Reply:
sort of, if at any point an error occurred it returned whatever it had, with the new setup though with validation being done at the start it works like totals and components with tpc marker = N being the equivalent of the STOP marker, we validate, throw an error if theres an issue same with T&C and if after that the method decides the data is not viable for correction it marks it as TPC = N and skips the rest of the steps